apps/react-storybook/stories/chat/Chat.stories.tsx

```diff
@@ -477,3 +491,109 @@ export const TypingUsers: Story = {
+            options: [
+                'No alerts',
+                'Limit reached',
+            ],
+            mapping: {
+                ['No alerts']: [],
+                ['Limit reached']: [{ message: 'Request limit reached, try again in a minute.' }],
+            },
+        },
+    },
+    render: ({
+                 width,
+                 height,
+                 alerts,
+             }) => {
+        const [isProcessing, setIsProcessing] = useState(false);
+        const [assistantReplyIndex, setAssistantReplyIndex] = useState<number>(0);
+        const [copyBtnIcon, setCopyBtnIcon] = useState('copy');
+
+        const items = useMemo(() => {
+            const repliesCount = assistantReplies.length;
+            const assistantReply = isProcessing ? regenerationMessage : assistantReplies[assistantReplyIndex % repliesCount];
+
+            return [ userRequest, assistantReply ];
+        }, [assistantReplyIndex, isProcessing]);
+
+        const onRegenerateButtonClick = useCallback(async (): Promise<void> => {
+            setIsProcessing(true);
+
+            await new Promise((resolve) => {
+                setTimeout(resolve, 300);
+            });
+            setAssistantReplyIndex((prev: number) => prev + 1);
+
+            setIsProcessing(false);
+        }, []);
+
+        const messageRender = useCallback(({ message }: { message: ChatTypes.Message }) => {
+            const { text = '', author } = message;
+
+            const onCopyButtonClick = () => {
+                navigator.clipboard?.writeText(text);
+                setCopyBtnIcon('check');
+
+                setTimeout(() => {
+                    setCopyBtnIcon('copy');
+                }, 2500);
+            };
+
+            if (text === REGENERATION_TEXT || author !== assistant) {
+                return <span>{text}</span>;
+            }
+
+            return (
+                <>
+                    <div>
+                        {HTMLReactParser(text)}
+                    </div>
+                    <div>
+                        <Button
+                            icon={copyBtnIcon}
+                            stylingMode='text'
+                            hint='Copy'
+                            onClick={onCopyButtonClick}
+                        />
+                        <Button
+                            icon='refresh'
+                            stylingMode='text'
+                            hint='Regenerate'
+                            onClick={onRegenerateButtonClick}
+                        />
+                    </div>
+                </>
+            );
+        }, [copyBtnIcon, onRegenerateButtonClick]);
+
+        return (
+            <div style={{ display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
+                <Chat
+                    height={height}
+                    width={width}
+                    items={items}
+                    reloadOnChange={false}
```

Comment:
We don't use DataSource object. I assume that this property doesn't matter for this sample.